### PR TITLE
Add Admin API

### PR DESCRIPTION
**Pull request by murtazamehmudji**
Added Add Category API for Admin

### CODE REVIEW

**Review by bityogi — APPROVED** [review]

**Review by miladezzat — CHANGES_REQUESTED** [review]
Where is the test cases

**Review by miladezzat — CHANGES_REQUESTED** [review]

File: src/admin/admin-category/admin-category.service.ts [hunk 1]
```diff
@@ -1,19 +1,42 @@
...
+    @InjectModel(Category.name)
+    private readonly categoryModel: Model<Category>,
+  ) {}
+
+  async create(
+    createAdminCategoryInput: CreateAdminCategoryInput,
+  ): Promise<Category> {
+    const category = await this.categoryModel.findOne({
+      name: createAdminCategoryInput.name,
+    });
```
Comment (miladezzat) [inline src/admin/admin-category/admin-category.service.ts:24]:
```suggestion
    const category = await this.categoryModel.findOne({
      name: createAdminCategoryInput.name, isDeleted: false
    });
```

File: src/admin/admin-category/admin-category.service.ts [hunk 2]
```diff
@@ -1,19 +1,42 @@
...
+    console.log(category);
+
+    if (category) {
+      throw new ConflictException('Category already exist');
+    }
+
+    return await this.categoryModel.create(createAdminCategoryInput);
```
Comment (miladezzat) [inline src/admin/admin-category/admin-category.service.ts:31]:
```suggestion
    return  this.categoryModel.create(createAdminCategoryInput);
```

File: src/admin/admin-category/admin-category.service.ts [hunk 3]
```diff
@@ -1,19 +1,42 @@
...
   }
 
-  findAll() {
-    return `This action returns all adminCategory`;
+  async findAll() {
+    return await this.categoryModel.find();
```
Comment (miladezzat) [inline src/admin/admin-category/admin-category.service.ts:35]:
```suggestion
    return  this.categoryModel.find({ isDeleted: false });
```

File: src/admin/admin-category/admin-category.service.ts [hunk 4]
```diff
@@ -1,19 +1,42 @@
...
   }
 
-  findOne(id: number) {
-    return `This action returns a #${id} adminCategory`;
+  async findOne(id: string) {
+    return await this.categoryModel.findOne({ _id: id });
```
Comment (miladezzat) [inline src/admin/admin-category/admin-category.service.ts:39]:
```suggestion
    return this.categoryModel.findOne({ _id: id, isDeleted: false });
```

**Pull request merged**